security/advancedtls/advancedtls.go

```diff
@@ -115,27 +160,11 @@ const (
 	SkipVerification
 )
 
-// ClientOptions contains all the fields and functions needed to be filled by
-// the client.
-// General rules for certificate setting on client side:
-// Certificates or GetClientCertificate indicates the certificates sent from
-// the client to the server to prove client's identities. The rules for setting
-// these two fields are:
-// If requiring mutual authentication on server side:
-//     Either Certificates or GetClientCertificate must be set; the other will
-//     be ignored.
-// Otherwise:
-//     Nothing needed(the two fields will be ignored).
+// ClientOptions contains the fields needed to be filled by the client.
 type ClientOptions struct {
-	// If field Certificates is set, field GetClientCertificate will be ignored.
-	// The client will use Certificates every time when asked for a certificate,
-	// without performing certificate reloading.
-	Certificates []tls.Certificate
-	// If GetClientCertificate is set and Certificates is nil, the client will
-	// invoke this function every time asked to present certificates to the
-	// server when a new connection is established. This is known as peer
-	// certificate reloading.
-	GetClientCertificate func(*tls.CertificateRequestInfo) (*tls.Certificate, error)
+	// IdentityCertificateOptions is OPTIONAL on client side. This field only
+	// needs to be set if mutual authentication is required on server side.
+	IdentityCertificateOptions
```

Comment:
Why is this embedding required? Just saving a few characters which referring to these options does not seem like a good enough reason to me, especially since it hurts readability of the code. This is especially more pronounced in the `RootCertificateOptions` case since it is not even at the top of the struct, so its not very obvious.

Reply:
Yeah that's a fair suggestion. Removed the embedded structure.